src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -730,7 +730,7 @@ private sealed class Mapper : MapperBase
                     var originalShape = _parent.TFInputShapes[i];
                     var shape = originalShape.ToIntArray();
 
-                    var colTypeDims = vecType.Dimensions.Select(dim => (long)dim).ToArray();
+                    var colTypeDims = vecType.Dimensions.Prepend(1).Select(dim => (long)dim).ToArray();
                     if (shape == null)
```

Comment:
This is not right. What if the expected shape is really 3-D? In addition, tensorflow internally can do reshape. Why can't we put the weight on tensorflow's shoulder?

Reply:
Well, it doesn't. At least for image models we use.
I'm actually not sure why, considering what usually models have variable tensors like [?,224,224,3], but in one which samples has problems they have [1,224,224,3]

Reply:
Then we need to fix those models or edit them before their exestuation. It's not idea to have an assumption which is true only for some image models.

Reply:
@Ivanidzo4ka, I agree with @wschin. It will create problems for models that don't have batch dimension. There were a couple of issues related to that on github. I already talked to @shauheen and @CESARDELATORRE on this and proposed following solution.

`The ultimate solution is to have a reshape transform that user can use to reshape their data. However, the temporary solution would be to add a parameter in options class called “AddBatchDimensionOnInput”. When user set it to true, batch dimension would be added to the inputs otherwise not.`

Can you instead do it for now in this PR? We can then discuss adding reshape transform later.

Reply:
This is the inception model from google.
https://storage.googleapis.com/download.tensorflow.org/models/inception5h.zip

Which takes input data of any shape. So we cannot infer shape from the input. There is a convolution layer just after the input which requires 4-D input and that's the problem.